File: 16/numOfOnes.c
/*
    @author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
    Description of problem at: https://www.codeeval.com/open_challenges/16/
*/

#include <stdio.h>
#include <string.h>

#define LINE_SIZE 1024

int main(int argc, char **argv) {
    FILE *ifp = fopen(argv[1], "r");
    int decNum = 0;
    while(fscanf(ifp, "%d\n", &decNum) != EOF) {
		int i = 0, j = 0, counter1 = 0, n = 0;
		if (decNum == 1) {
			printf("%d\n", 1);
			continue;
		}
		if (decNum == 0) {
			printf("%d\n", 0);
			continue;
		}
        for (i = 0; decNum > 0; i++) {
			n = decNum % 2;
			if (n == 1)
				counter1++;
			decNum = decNum / 2;
		}
		printf("%d\n", counter1);
    }
    system("pause");
    return 0;
}

File: 016/numOfOnes.java
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/16/
*/

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

public class numOfOnes {
	public static void main(String[] args) {
		try {
			File file = new File(args[0]);
			BufferedReader infile = new BufferedReader(new FileReader(file));
			String line;
			while ((line = infile.readLine()) != null) {
				int num = Integer.parseInt(line);
				String binary = Integer.toBinaryString(num);
				int total = 0;
				for (int i = 0; i < binary.length(); i++) {
					char c = binary.charAt(i);
					String cc = Character.toString(c);
					num = Integer.parseInt(cc);
					if (num == 1) {
						total++;
					}
				}
				System.out.println(total);
			}
			infile.close();
		} catch (FileNotFoundException e) {
			System.out.println("This is crap. Wrong stupid file name.");
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}